app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,57 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('config_info'))
+      config_info = data['config_info'].except('ae_endpoints') || {}
+      case service_template.type
+      when 'ServiceTemplateOrchestration'
```

Comment:
We are going to add new type `ServiceTemplateContainer`, and possibly other types in the future. How do we remember to come back to update this `case...when`? 
The API controller attempts to create a service template, but the actual ServiceTemplate has many subclasses. Need a generic way to handle subtypes.